### DISCUSSION TITLE
Folder Processing with process-folder.js

### DISCUSSION THREAD

**Discussion by saeidn95** [vladmandic/human#448] [Q&A]
I see an unexpected behavior with "process-folder.js" 

From what I see after the first image,
a.   images are loaded in batches of 4. But then they processed one at a time with a box output  produced after the processing of each image by the blazeface network.
b. What is unusual is that  results of blazeface network does not get processed immediately by the facemesh network.  A batch of 4 boxes from the blazeface is collected first before they are processed by the batch-wise by the facemesh network (https://github.com/vladmandic/human/blob/62396317f5c31b2ec5bc7de2e9eb53b8978b9f28/src/face/facemesh.ts#L92C6-L92C72), Is there a reason for this? and can I configure this batch processing to do it one at a time.

**Comment by vladmandic** [comment]
everything in `demo` folder are just demos - they are not even intended to be immediately used as-is, so modify them as much as you like.
regarding batch size of 4, that's just the default and set as constant.

**Reply by saeidn95** [reply]
The question I had really was about the reason for one wanting to do processing  batch-wise. Does it improve the processing speed? or there is something else in play here. 

BTW: where is the location of  the parameter that sets the batch size?

**Comment by vladmandic** [comment]
https://github.com/vladmandic/human/blob/62396317f5c31b2ec5bc7de2e9eb53b8978b9f28/demo/nodejs/process-folder.js#L31

**Reply by vladmandic** [reply]
and no, it doesn't change processing speed that much. its a demo so you can see how such functionality can work, nothing else.
